src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/JoinableTaskExtensions.cs

```diff
@@ -22,6 +23,10 @@ public static void PostOnFailure(this JoinableTask joinableTask, string callerCl
 #pragma warning disable VSTHRD003 // Avoid awaiting foreign Tasks - As a fire-and-forget continuation, deadlocks can't happen.
                     await joinableTask.Task.ConfigureAwait(false);
 #pragma warning restore VSTHRD003
+                }
+                catch (TaskCanceledException taskCanceledException)
```

Comment:
What about the base class? If I understand, `ThrowIfCancellationRequested` will  use
[OperationCanceledException type](https://referencesource.microsoft.com/#mscorlib/system/threading/CancellationToken.cs,466) and will `throw new OperationCanceledException` [(code)](https://referencesource.microsoft.com/#mscorlib/system/threading/CancellationToken.cs,482)